infra/terraform/global/iam.tf

```diff
@@ -0,0 +1,27 @@
+resource "aws_iam_user" "auth0_ses_user" {
+    name = "auth0-ses-user"
+}
+
+resource "aws_iam_user_policy" "auth0-ses-user" {
```

Comment:
imo that's not necessary, as you always need to refer to Terraform resources by both type and name - always `aws_iam_user_policy.auth0-ses-user`, never just `auth0-ses-user`. That said, the names for these resources should probably be the more general `auth0-ses` rather than referring to any particular resource.